x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java

```diff
@@ -355,6 +357,8 @@ protected void onStart(long now, ActionListener<Boolean> listener) {
             }));
         } else {
             hasSourceChanged = true;
+            context.setChangesLastSearchedAt(instantOfTrigger);
+            context.setChangesLastDetectedAt(instantOfTrigger);
```

Comment:
This changes behaviour for `changed_last_detected_at`. The change seems correct but is it fixing another issue with that not getting updated properly?

Reply:
yes, the `if` is "continuous" mode, the `else` is "batch". I see no reason why we report  `changed_last_detected_at` (and `changed_last_searched_at`) only for continuous but not for batch. This makes it more consistent.